### ISSUE TITLE
[BUG] Bluetooth Controller Misbehaving

### ISSUE THREAD

**Issue by hermetic-charm** [linuxserver/docker-emulatorjs#72]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Bluetooth controller is connected but either does nothing or goes "crazy" looping through menus.

### Expected Behavior

Bluetooth controller works in both the menus and games.

### Steps To Reproduce

1. Open game, controller does nothing
2. Check Retroarch Menu (F1)->Controls->Port 1 Controls: It can see the controller it's listed under buttons, controller still does nothing
![image](https://github.com/linuxserver/docker-emulatorjs/assets/11345378/2ce060ab-72d8-45cb-a08d-9fdd5f8e8683)
4. Select any button menu item, e.g. D-pad Up, to open the next screen which shows button selection options.
5. Press any button on controller (e.g. A, B, Dpad Up): Controller input gets recognized but now input starts going crazy like I am holding down the Dpad button.  It scrolls through menu options infinitely.  Oddly, while it is going crazy other buttons seem to work (e.g. I can press B or A and it will go into/out of a menu item as it scrolls through everything).
6. Pressing F1 gets me out of the crazy menus and back into the game, but now the controller does nothing again
7. Additionally, now after the above steps, trying to go back into F1 menu nothing works.  I can't do anything, keyboard does not work, mouse does not work, gamepad does not work.
8. Refresh page.  Still does not work.
9. The only way to get control of the menu again is to unpair the Bluetooth controller.  I see a message toast in the browser saying controller has been disconnected and then menu functionality comes back with mouse/keyboard.

### Environment

```markdown
- OS: Windows 11
- How docker service was installed: Synology Container Manager
- Browser: Edge
-Controller: 8bitdo NES controller connected via Bluetooth.  Browser sees controller as confirmed on https://gamepadtest.com/
```

### CPU architecture

x86-64

### Docker creation

```bash
services:
  emulatorjs:
    image: lscr.io/linuxserver/emulatorjs:latest
    container_name: emulatorjs
    environment:
      - PUID=REDACTED
      - PGID=REDACTED
      - TZ=America/New_York
      #- SUBFOLDER=/ #optional
    security_opt:
      - no-new-privileges:true
    network_mode: bridge
    volumes:
      - /path/tomy/config:/config
      - /path/tomy/data:/data
    ports:
      - REDACTED:3000
      - REDACTED:80
      #- REDACTED:4001 #optional
    restart: on-failure:20
```

### Container logs

```bash
[migrations] started
[migrations] no migrations found
───────────────────────────────────────

      ██╗     ███████╗██╗ ██████╗
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝

   Brought to you by linuxserver.io
───────────────────────────────────────

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    REDACTED
User GID:    REDACTED
───────────────────────────────────────

[ipfs-upgrade] Checking if fs-repo needs to be upgraded (this may take some time)
[custom-init] No custom files found, skipping...
Initializing daemon...
Kubo version: 0.24.0
Repo version: 15
System version: amd64/linux
Golang version: go1.21.3
[ls.io-init] done.
Swarm listening on /ip4/127.0.0.1/tcp/4001
Swarm listening on /ip4/127.0.0.1/udp/4001/quic-v1
Swarm listening on /ip4/127.0.0.1/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm listening on /ip4/IP_REDACTED/tcp/4001
Swarm listening on /ip4/IP_REDACTED/udp/4001/quic-v1
Swarm listening on /ip4/IP_REDACTED/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm listening on /p2p-circuit
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm announcing /ip4/127.0.0.1/tcp/4001
Swarm announcing /ip4/127.0.0.1/udp/4001/quic-v1
Swarm announcing /ip4/127.0.0.1/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm announcing /ip4/IP_REDACTED/tcp/4001
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
RPC API server listening on /ip4/127.0.0.1/tcp/5001
WebUI: http://127.0.0.1:5001/webui
Gateway server listening on /ip4/127.0.0.1/tcp/8080
Daemon is ready

Received interrupt signal, shutting down...
(Hit ctrl-c again to force-shutdown the daemon.)
[migrations] started
[migrations] no migrations found
usermod: no changes
───────────────────────────────────────

      ██╗     ███████╗██╗ ██████╗
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝

   Brought to you by linuxserver.io
───────────────────────────────────────

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    REDACTED
User GID:    REDACTED
───────────────────────────────────────

[ipfs-upgrade] Checking if fs-repo needs to be upgraded (this may take some time)
[custom-init] No custom files found, skipping...
Initializing daemon...
Kubo version: 0.24.0
Repo version: 15
System version: amd64/linux
Golang version: go1.21.3
Swarm listening on /ip4/127.0.0.1/tcp/4001
Swarm listening on /ip4/127.0.0.1/udp/4001/quic-v1
Swarm listening on /ip4/127.0.0.1/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm listening on /ip4/IP_REDACTED/tcp/4001
Swarm listening on /ip4/IP_REDACTED/udp/4001/quic-v1
Swarm listening on /ip4/IP_REDACTED/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm listening on /p2p-circuit
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm announcing /ip4/127.0.0.1/tcp/4001
Swarm announcing /ip4/127.0.0.1/udp/4001/quic-v1
Swarm announcing /ip4/127.0.0.1/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
Swarm announcing /ip4/IP_REDACTED/tcp/4001
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1
Swarm announcing /ip4/IP_REDACTED/udp/4001/quic-v1/webtransport/certhash/HASH_REDACTED/certhash/HASH_REDACTED
RPC API server listening on /ip4/127.0.0.1/tcp/5001
WebUI: http://127.0.0.1:5001/webui
Gateway server listening on /ip4/127.0.0.1/tcp/8080
Daemon is ready
[ls.io-init] done.
```

**Comment by thelamer** [comment]
The only controllers I have to test are Xbox series, Xbox wired(360 and wired), Razr kishi
Though I can say that low button controllers likely crash the menu as this logic needs a 16th button to detect triple home button presses to drop back to the menu.
https://github.com/linuxserver/emulatorjs/blob/master/frontend/js/index.js#L579
Also are you sure this is Bluetooth and not 2.4 ghz ?

Check the browser console for errors.

**Comment by thelamer** [comment]
Just to be clear when I say to check the browser console for errors those are the errors I need to know what is going wrong.

**Comment by thelamer** [comment]
The error should be when you are in the game browser if the controller does not work there.

**Comment by thelamer** [comment]
Does this page work for you ? 
https://hardwaretester.com/gamepad

**Comment by thelamer** [comment]
I think I can work with that, i have a lot on my plate right now will take a bit to get you something to test.

**Comment by thelamer** [comment]
@hermetic-charm Can you please test this code?

Exec into the container to replace the file with head: 

```
docker exec emulatorjs sh -c 'curl -L https://raw.githubusercontent.com/linuxserver/emulatorjs/master/frontend/js/index.js -o /emulatorjs/frontend/js/index.js'
```

Then in your browser use a private window to make sure you don't get any cache and see if the controller works at all in the menu or in game.

**Comment by thelamer** [comment]
It looks like your controller dpad is registered as n analogue device, I will get a fix in for the up down not working in the menu.
The other errors are because you have to click on the page chrome does not see the controller input as interaction it seems anymore.

As for gameplay in retroarch itself you need to set I believe:
```
Settings, Input, User 1 Binds, set User 1 Analog to Digital Type to Left Analog. 
```
Keep in mind retroarch thinks your dpad is an analogue input.

**Comment by thelamer** [comment]
Run this again this should fix the menu: 
```
docker exec emulatorjs sh -c 'curl -L https://raw.githubusercontent.com/linuxserver/emulatorjs/master/frontend/js/index.js -o /emulatorjs/frontend/js/index.js'
```

And again I think you need to interact with the page (mouse click into it or keyboard) before you start doing controller stuff or at the very least before trying to launch a game. That last error is from Retroarch itself and I have never seen it though I doubt it is related to  a controller.

**Comment by thelamer** [comment]
Can you show me a screenshot of what it looks like when you press down or up on your controller from here https://hardwaretester.com/gamepad ?

**Comment by thelamer** [comment]
So you see the problem here right? 
It's two fold, you have a dead zone registered on the controller that `-0.00952` registered with no input (it should be 0). That is fine and we work around that by only registering analog presses of .5 and greater. But you also have these axis 3 and 4 that are permanently pressed. That is not something I believe Retroarch can deal with and I do not want to shim anything in to blacklist in the frontend as it does not fix retroarch and axis 3 is the left analog on all normal controllers and it can be the left analog on many others. 

Basically everything should be registered as 0 as seen with my series controller here: 

![controller](https://github.com/linuxserver/docker-emulatorjs/assets/1852688/09ee7a6c-feb0-4d23-9cc1-fdce936cbb70)


Your only real course of action is to contact 8bitdo, they need to work to get their controller sane in web browsers and participate in its mapping and identification.
